### PR TITLE
dynamic mask support

### PR DESCRIPTION
**Pull request by bo-dun-1**

### CODE REVIEW

File: src/form/Form.jsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ import { getPlaidFieldValues, openPlaidLink } from '../integrations/plaid';
 import Spinner from 'react-bootstrap/Spinner';
 import Lottie from 'lottie-react';
 import { useHotkeys } from 'react-hotkeys-hook';
+import { getTextFieldMask } from '../elements/fields/TextField';
```
Comment (jakeandbakelindvall) [inline src/form/Form.jsx:46]: unused?

File: src/elements/fields/TextField.jsx [hunk 2]
```diff
@@ -1,15 +1,47 @@
...
+      return 'b';
+    case 'digit':
+      return '0';
+    default:
+      return '*';
+  }
+}
+
+function getTextFieldMask(servar) {
+  const prefix = escapeDefinitionChars(servar.metadata.prefix || 'asd');
```
Comment (jakeandbakelindvall) [inline src/elements/fields/TextField.jsx:32]: quick explanation for `asd`?

File: src/elements/fields/TextField.jsx [hunk 3]
```diff
@@ -1,15 +1,47 @@
...
+
 import InlineTooltip from '../components/Tooltip';
 import { bootstrapStyles } from '../styles';
 import { emailPatternStr } from '../../utils/formHelperFunctions';
 
-function getTextFieldProps(servar, styles, value) {
-  let methods, onlyPhone;
+const MAX_TEXT_FIELD_LENGTH = 512;
+
+function escapeDefinitionChars(str) {
```
Comment (jakeandbakelindvall) [inline src/elements/fields/TextField.jsx:10]: reason for this?

File: src/elements/fields/TextField.jsx [hunk 4]
```diff
@@ -80,17 +148,20 @@ function TextField({
...
   elementProps = {},
   inlineError,
-  inputRef,
-  ...fieldProps
+  rawValue = '',
+  onAccept = () => {}
 }) {
   const servar = element.servar;
-  const inputType = fieldProps.as === 'textarea' ? 'textarea' : 'input';
+  const maskProps = {
```
Comment (jakeandbakelindvall) [inline src/elements/fields/TextField.jsx:159]: reason for spreading and then wrapping in object? function call should be enough right?

**Review by jakeandbakelindvall — COMMENTED** [review]
looking good

File: src/form/Form.jsx (hunk 1 above)
Comment (bo-dun-1) [reply src/form/Form.jsx:46]: good catch

File: src/elements/fields/TextField.jsx (hunk 2 above)
Comment (bo-dun-1) [reply src/elements/fields/TextField.jsx:32]: oh jeez, i need to remove this

File: src/elements/fields/TextField.jsx (hunk 3 above)
Comment (bo-dun-1) [reply src/elements/fields/TextField.jsx:10]: some imask characters are special, e.g. * means any char allowed, 0 means any digit allowed, etc. this makes them treated literally

File: src/elements/fields/TextField.jsx (hunk 4 above)
Comment (bo-dun-1) [reply src/elements/fields/TextField.jsx:159]: yes you're right

**Pull request merged**